lib/msf/core/modules/metadata/search.rb

```diff
@@ -6,14 +6,14 @@
 module Msf::Modules::Metadata::Search
 
   VALID_PARAMS =
-      %w[app author authors arch cve bid edb date disclosure_date description full_name fullname mod_time name
-        os platform path port rport rank ref ref_name reference references target targets text type]
+      %w[app author authors arch cve bid edb check date disclosure_date description full_name fullname mod_time
+      name os platform path port rport rank ref ref_name reference references target targets text type]
 
   #
   # Searches the module metadata using the passed hash of search params
   #
   def find(params, fields={})
-    return [] if params.empty? || VALID_PARAMS.none? { |k| params.key?(k) }
+    return nil if params.empty? || VALID_PARAMS.none? { |k| params.key?(k) }
```

Comment:
Is there a specific reason why we're returning `nil` on invalid params instead of raising an `ArgumentError`?